Userland/Services/WebContent/WebContentServer.ipc

```diff
@@ -20,4 +20,6 @@ endpoint WebContentServer = 89
     KeyDown(i32 key, unsigned modifiers, u32 code_point) =|
 
     DebugRequest(String request, String argument) =|
+
+    GetSource() => (String source)
```

Comment:
All IPC API's provided by WebContentServer should be async (`=|`) so the Browser never has to block while waiting for a response :)

Reply:
Is getting the source going to be a blocking operation though? Shouldn't it return null If the document isn't loaded yet?

Reply:
Since it's done over IPC, we have no guarantees that the request will be serviced immediately. Hence we must be asynchronous or risk blocking the browser GUI

Reply:
So would I create method to request the sourcecode, and then have something like DidRetrievedSourcecode which would get fired back from the handle method?

Reply:
Right :)

Reply:
Everything should be async now.